src/jshint.js

```diff
@@ -4540,13 +4540,14 @@ var JSHINT = (function() {
   // expression is a comprehension array, destructuring assignment or a json value.
 
   var lookupBlockType = function() {
-    var pn, pn1;
+    var pn, pn1, prev;
     var i = -1;
     var bracketStack = 0;
     var ret = {};
     if (checkPunctuators(state.tokens.curr, ["[", "{"]))
       bracketStack += 1;
     do {
+      prev = pn || state.tokens.curr;
       pn = (i === -1) ? state.tokens.next : peek(i);
```

Comment:
how about ?
```
if (i === -1) {
    prev = state.tokens.curr;
    pn = state.tokens.next;
} else {
   prev = pn;
   pn = peek(i);
}
```
or making both the inline conditionals the same?
```
prev = i === -1 ? state.tokens.curr : pn;
```
at the moment they use a different method to determine is first and they are reversed conditionals!